codex/merkletree/merkletree.nim

```diff
@@ -1,96 +1,136 @@
 ## Nim-Codex
-## Copyright (c) 2022 Status Research & Development GmbH
+## Copyright (c) 2023 Status Research & Development GmbH
 ## Licensed under either of
 ##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
 ##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
 ## at your option.
 ## This file may not be copied, modified, or distributed except according to
 ## those terms.
 
-import std/sequtils
 import std/math
 import std/bitops
-import std/sugar
+import std/strutils
 
-import pkg/libp2p
-import pkg/stew/byteutils
-import pkg/questionable
 import pkg/questionable/results
+import pkg/nimcrypto/sha2
+
+const digestSize = sha256.sizeDigest
 
 type
-  MerkleHash* = MultiHash
+  MerkleHash* = array[digestSize, byte]
```

Comment:
We still want to use `Multihash` for the interior nodes, we only want to allow generic byte streams for leaf nodes.

Reply:
Fixed in the latest batch of changes